### PR TITLE
[FLINK-18140][doc][orc] Add documentation for ORC format

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

Add documentation for ORC format for filesystem connector.

## Verifying this change

image
image

### CODE REVIEW

File: docs/dev/table/connectors/formats/orc.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+Dependencies
+------------
+
+In order to setup the Orc format, the following table provides dependency information for both
+projects using a build automation tool (such as Maven or SBT) and SQL Client with SQL JAR bundles.
+
+| Maven dependency   | SQL Client JAR         |
+| :----------------- | :----------------------|
+| `flink-orc`        | [Download](https://repo.maven.apache.org/maven2/org/apache/flink/flink-orc{{site.scala_version_suffix}}/{{site.version}}/flink-orc{{site.scala_version_suffix}}-{{site.version}}-jar-with-dependencies.jar) |
+
```
Comment (danny0405) [inline docs/dev/table/connectors/formats/orc.md:43]: Add `{% if site.is_stable %}` to show the link for stable release ?

File: docs/dev/table/connectors/formats/orc.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    </tr>
+    <tr>
+      <td>TIMESTAMP</td>
+      <td>timestamp</td>
+      <td>TIMESTAMP</td>
+    </tr>
+    </tbody>
+</table>
+
+<span class="label label-danger">Attention</span> Not support complex data type like: Array, Map, Row.
```
Comment (danny0405) [inline docs/dev/table/connectors/formats/orc.md:183]: Composite data type: `Array`, `Map` and `Row` are not supported.

File: docs/dev/table/connectors/formats/orc.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+Dependencies
+------------
+
+In order to setup the Orc format, the following table provides dependency information for both
+projects using a build automation tool (such as Maven or SBT) and SQL Client with SQL JAR bundles.
+
+| Maven dependency   | SQL Client JAR         |
+| :----------------- | :----------------------|
+| `flink-orc`        |{% if site.is_stable %} [Download](https://repo.maven.apache.org/maven2/org/apache/flink/flink-orc{{site.scala_version_suffix}}/{{site.version}}/flink-orc{{site.scala_version_suffix}}-{{site.version}}-jar-with-dependencies.jar) {% else %} Only available for stable releases {% endif %}|
```
Comment (lirui-apache) [inline docs/dev/table/connectors/formats/orc.md:42]: I think it's better to provide the full coordinates of the maven artifact, e.g. `org.apache.flink:flink-orc...`. Or maybe something like
```
		<dependency>
			<groupId>org.apache.flink</groupId>
			<artifactId>flink-orc_{{site.scala_version_suffix}}</artifactId>
			<version>{{site.version}}</version>
		</dependency>
```
Then users can simply copy it to their pom files.

File: docs/dev/table/connectors/formats/orc.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Specify what format to use, here should be 'orc'.</td>
+    </tr>
+    </tbody>
+</table>
+
+Orc format also supports table properties from [Table properties](https://orc.apache.org/docs/hive-config.html#table-properties).
+For example, you can configure `orc.compress=SNAPPY` to enable snappy compression.
```
Comment (lirui-apache) [inline docs/dev/table/connectors/formats/orc.md:93]: Shall we list all the supported properties? Or maybe we can add link to Orc doc if all the properties are supported.

File: docs/dev/table/connectors/formats/orc.md (hunk 3 above)
Comment (JingsongLi) [reply docs/dev/table/connectors/formats/orc.md:42]: It is currently Flink dependency style.

File: docs/dev/table/connectors/formats/orc.md (hunk 4 above)
Comment (JingsongLi) [reply docs/dev/table/connectors/formats/orc.md:93]: We don't need add all orc properties. I have added Orc doc.

File: docs/dev/table/connectors/formats/orc.md (hunk 3 above)
Comment (JingsongLi) [reply docs/dev/table/connectors/formats/orc.md:42]: But I think we can add `_{{site.scala_version_suffix}}`

**Comment by JingsongLi** [comment]
Merging...

**Pull request merged**